overlord/ifacestate/handlers.go

```diff
@@ -232,17 +235,60 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	plug := m.repo.Plug(plugRef.Snap, plugRef.Name)
+	if plug == nil {
+		return fmt.Errorf("no such plug")
```

Comment:
"snap %q has no %q plug"?

We'll have issues here, though. This will conflict with the improvements we have in progress for connect/disconnect, which support doing that implicitly. We'll probably need to separate the resolution of plugs and slots from the actual connection disconnection.

That seems like a good idea anyway. Let's move this on and keep the fixes for the other branch.